modules/stochastic_tools/include/utils/LibtorchNeuralNetBase.h

```diff
@@ -0,0 +1,41 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#ifdef TORCH_ENABLED
+#include <torch/torch.h>
+#include "MooseError.h"
+
+namespace StochasticTools
+{
+
+// This base class is meant to gather the functions and members common in
+// every libtorch-based class. Note, it already inherits from the torch::nn::Module
+// so this will manage the layers (submodules) of the derived classes as well.
+class LibtorchNeuralNetBase : public torch::nn::Module
+{
+public:
+  // Generic constructor
```

Comment:
Is this necessary?

Reply:
Actually, I think I'll move this guy and the derived classes to the framework/utils since this coulc be reused practically anywhere.